File: 2022/a.py
#!/usr/bin/env python3

# Problem A - Counting Peaks of Infection

def solve(v):
    return sum(1 for a, b, c in zip(v, v[1:], v[2:]) if a < b and c < b)

while True:
    n = int(input())
    if n == 0:
        break
    v = list(map(int, input().split()))
    assert len(v) == n
    print(solve(v))

File: 2022/a.ex
#!/usr/bin/env elixir

# ACM-ICPC 2022 Domestic Qualifier
# Problem A - Counting Peaks of Infection

defmodule Main do
  def main() do
    n = IO.gets("") |> String.trim() |> String.to_integer()
    if n != 0 do
      v = IO.gets("") |> String.split(~r{\s}, trim: true)
            |> Enum.map(&String.to_integer/1)
      solve(n, v) |> IO.inspect()
      main()
    end
  end

  def solve(_n, v) do
    Enum.zip([v, tl(v), tl(tl(v))])
      |> Enum.filter(fn {a, b, c} -> a < b && b > c end)
      |> length()
  end
end

Main.main()
